shell/platform/tizen/flutter_tizen_shell.cc

```diff
@@ -1,67 +1,58 @@
-#include <assert.h>
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
 #include <unistd.h>
-#include <chrono>
-#include <iostream>
-#include <thread>
+
+#include <string>
+#include <vector>
 
 #include "flutter/shell/platform/tizen/flutter_tizen_engine.h"
 #include "flutter/shell/platform/tizen/public/flutter_tizen.h"
 
-extern int gApp_width;
-extern int gApp_height;
-
-std::string TPK_ROOT_PATH = "/tpkroot";
-std::string LIB_PATH = "/lib";
-std::string RES_PATH = "/res";
-
 class FlutterApp {
  public:
   explicit FlutterApp() {}
   virtual ~FlutterApp() {}
 
   bool OnCreate() {
     printf("Launching a Flutter application...\n");
-    std::string assets_path;
-    std::string icu_data_path;
-    std::string aot_lib_path;
-    FlutterDesktopEngineProperties engine_prop = {};
-    std::vector<const char*> switches;
 
-    std::string tpk_root;
-    if (app_path_.empty()) {
-      char path[256];
-      getcwd(path, sizeof(path));
-      tpk_root = path + std::string("/tpkroot");
-    } else {
-      tpk_root = app_path_;
-    }
+    FlutterDesktopWindowProperties window_prop = {};
+    window_prop.headed = true;
+    window_prop.x = 0;
+    window_prop.y = 0;
+    window_prop.width = window_width_;
+    window_prop.height = window_height_;
 
-    assets_path = tpk_root + "/res/flutter_assets";
-    icu_data_path = tpk_root + "/res/icudtl.dat";
-    aot_lib_path = tpk_root + "/lib/libapp.so";
+    std::string assets_path = app_path_ + "/res/flutter_assets";
+    std::string icu_data_path = app_path_ + "/res/icudtl.dat";
+    std::string aot_lib_path = app_path_ + "/lib/libapp.so";
 
+    std::vector<const char*> switches;
     switches.push_back("--disable-observatory");
     switches.push_back("--verbose-logging");
     switches.push_back("--enable-dart-profiling");
     switches.push_back("--enable-checked-mode");
 
+    FlutterDesktopEngineProperties engine_prop = {};
     engine_prop.assets_path = assets_path.c_str();
     engine_prop.icu_data_path = icu_data_path.c_str();
     engine_prop.aot_library_path = aot_lib_path.c_str();
     engine_prop.switches = switches.data();
     engine_prop.switches_count = switches.size();
-    engine_ = FlutterDesktopRunEngine(engine_prop, true);
+    engine_ = FlutterDesktopRunEngine(window_prop, engine_prop);
 
     if (!engine_) {
       printf("Could not launch a Flutter application.\n");
       return false;
     }
-    // RegisterPlugins(this);
+
     return true;
   }
 
   void OnTerminate() {
-    printf("Shutting down the application...");
+    printf("Shutting down the application...\n");
```

Comment:
What do you think about modifying it as below and using the log macros?

``` diff
--- a/shell/platform/tizen/tizen_log_stub.cc
+++ b/shell/platform/tizen/tizen_log_stub.cc
@@ -19,7 +19,7 @@ namespace flutter {
 void SetMinLoggingLevel(log_priority p){};
 
 log_priority GetMinLoggingLevel() {
-  return DLOG_ERROR;
+  return DLOG_DEBUG;
 };
```

Reply:
As we're already passing `--verbose-logging` as an engine argument, there's no need to change `GetMinLoggingLevel`.

I don't think it makes sense to use `printf` in `OnCretae` and `FT_LOGI` in `OnTerminate`. I think it's better to leave them as they are or remove both from `OnCreate` and `OnTerminate`.

Reply:
Yes, you can remove both and it's better than the current.

Reply:
yes, we can remove both and it is nice rather then as it is